src/axis_builder.jl

```diff
@@ -17,6 +21,27 @@ Base.@kwdef mutable struct AxisStyle
     title = ""
 end
 
+#
+# We use Axis to draw the axis, in addition to the axisstyle.
+# Axis also contains information about the window:
+#
+#   width, height, windowbackgroundcolor, drawbackground,
+#
+# and information about the axis which is not style
+#
+#   ticks, box, yoriginatbottom
+#
+# and the AxisStyle object as. Note that the AxisStyle object
```

Comment:
"object as"? Oh, I see, you named the object `as`, but that makes this comment look not terminated. I suggest you either use Markdown to quote `as`, or rename to `style`.

Reply:
Also, is there an option to mark the style object as "const"? Essentially have the code reflect the invariant that style is unmodifiable.

Reply:
In Julia, one can mark the style object as const, but it would only prevent changing of the reference to it. If A is an Axis object, it would prevent changes such as A.as = new_as.  But it would not prevent changes such as A.as.title = "newtitle".

Reply:
I put the "as" in quotes in the comment. I don't think Julia supports Markdown in comments.